Add Haefele Connect Mesh brand icon and logo for custom integration

QNimbus opened the pull request:
## Proposed change

Adding brand and logo images for custom Häfele Connect Mesh integration.

## Type of change

- [ ] Add a new logo or icon for a new core integration
- [ ] Add a missing icon or logo for an existing core integration
- [x] Add a new logo or icon for a custom integration (custom component)
  - [x] I've added a link to my custom integration repository in the PR description
  - [ ] I've opened up a PR for my custom integration on the [Home Assistant
    Python wheels repository](https://github.com/home-assistant/wheels-custom-integrations)
- [ ] Replace an existing icon or logo with a higher quality version
- [ ] Replace an existing icon or logo after a branding change
- [ ] Removing an icon or logo

## Additional information

- Link to custom integration repository: https://github.com/[QNimbus/haefele-connect-mesh](https://github.com/QNimbus/haefele-connect-mesh) (Still under development)

## Checklist

- [x] The added/replaced image(s) are **PNG**
- [x] Icon image size is 256x256px (`icon.png`)
- [x] hDPI icon image size is 512x512px for  (`icon@2x.png`)
- [x] Logo image size has min 128px, but max 256px, on the shortest side (`logo.png`)
- [x] hDPI logo image size has min 256px, but max 512px, on the shortest side (`logo@2x.png`)

frenck reviewed (changes requested):
The branding provided in this PR looks different from the branding the manufacturer is using:

![CleanShot 2024-11-04 at 10 41 18@2x](https://github.com/user-attachments/assets/19fc5244-8f32-4399-a385-c50f61e91533)

![CleanShot 2024-11-04 at 10 41 27@2x](https://github.com/user-attachments/assets/9241eb42-7ff8-4eaf-9005-8b5dc7b2b24f)

Please make sure the branding matches the manufacturer branding.

../Frenck

QNimbus commented:
> The branding provided in this PR looks different from the branding the manufacturer is using:
> 
> ...
> 
> Please make sure the branding matches the manufacturer branding.
> 
> ../Frenck

I misunderstood initially - I added the Haefele Brand icon as `icon.png` and `icon@2x.png` and added the Haefele Connect Mesh product logo as `logo.png` and `logo@2x.png`. Thanks for your feedback!

frenck reviewed (changes requested):
>  and added the Haefele Connect Mesh product logo

We don't want to collect product branding in this repository (unless it is grouped in a brand, but that functionality isn't available for custom integrations).

../Frenck

frenck reviewed (changes requested):
Now the logos and icons are the same assets, we can omit the logo version. As per our README in this repository, we will automatically fall back to the icon if the logo is missing.

../Frenck
